runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -42,141 +52,291 @@
      * @return TxBuilder instance.
      */
     @Nonnull
     public <K extends Message, V extends Message, M extends Message>
-    TxBuilder touch(@Nonnull final String tableName,
-                    @Nonnull final K key) {
-        Table<K, V, M> table = getTable(tableName);
-        operations.add(() -> {
-            //TODO: Validate the get is executed.
-            CorfuRecord<V, M> record = table.get(key);
-        });
+    TxBuilder clear(@Nonnull Table<K, V, M> table) {
+        operations.add(table::clear);
         return this;
     }
 
     /**
      * Deletes the specified key.
      *
-     * @param tableName Table name to perform the delete on.
-     * @param key       Key.
+     * @param table Table object to perform the delete on.
+     * @param key       Key of the record to be deleted.
      * @param <K>       Type of Key.
      * @param <V>       Type of Value.
+     * @param <M>       Type of Metadata.
      * @return TxBuilder instance.
      */
     @Nonnull
     public <K extends Message, V extends Message, M extends Message>
-    TxBuilder delete(@Nonnull final String tableName,
+    TxBuilder delete(@Nonnull Table<K, V, M> table,
                      @Nonnull final K key) {
-        Table<K, V, M> table = getTable(tableName);
         operations.add(() -> table.delete(key));
         return this;
     }
 
-    private void txBegin() {
-        Transaction.TransactionBuilder transactionBuilder = this.objectsView
-                .TXBuild()
-                .type(TransactionType.OPTIMISTIC);
-        if (timestamp != null) {
-            transactionBuilder.snapshot(new Token(timestamp.getEpoch(), timestamp.getSequence()));
+    /**
+     *************************** READ API *****************************************
+     */
+
+    /**
+     * get the full record from the table given a key.
+     * If this is invoked on a Read-Your-Writes transaction, it will result in starting a corfu transaction
+     * and applying all the updates done so far.
+     *
+     * @param table Table object to retrieve the record from
+     * @param key   Key of the record.
+     * @return CorfuStoreEntry<Key, Value, Metadata> instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    CorfuStoreEntry getRecord(@Nonnull Table<K, V, M> table,
+                              @Nonnull final K key) {
+        if (readYourUncommittedWrites) {
+            txBeginInternal();
+            operations.forEach(Runnable::run); // Apply all pending mutations for reads to see it.
+            operations.clear();
+            CorfuRecord<V, M> record = table.get(key);
+            return new CorfuStoreEntry(key, record.getPayload(), record.getMetadata());
+        }
+        try {
+            txBeginInternal();
+            CorfuRecord<V, M> record = table.get(key);
+            return new CorfuStoreEntry(key, record.getPayload(), record.getMetadata());
+        } finally {
+            txEndInternal();
```

Comment:
this will not get executed in case if readYourUncommittedWrites==true, right?

Reply:
yes, the code returns before